src/main/java/com/tables/ST.java

```diff
@@ -187,13 +186,19 @@ private int rank(Key key, int lo, int hi) {
 			int mid = lo + (hi- lo) / 2;
 			int cmp = key.compareTo(keys[mid]);
 			
-			if (cmp > 0) {
+			if (cmp == 0)
+			   return mid;
+			else if (cmp > 0) {
 				lo = mid + 1;
 			}
 			else if (cmp < 0) {
 				hi = mid - 1;
 			}
-		}		
+		}
+		if (lo == 0)
```

Comment:
hi @JoshVandeWalle, can you explain me what do you trying to achieve with this logic?